test/Interop/Cxx/templates/Inputs/function-templates.h

```diff
@@ -44,6 +44,12 @@ decltype(auto) testAuto(T arg) {
   return arg;
 }
 
+template <typename T>
+T usedInDeclType(T) {}
+
+template <typename T>
+void takesDeclTypePointer(decltype(usedInDeclType<T>()) *) {}
+
```

Comment:
Optional: you can probably break this in a simpler way. I think this should fail without decltype. 
```suggestion
template <typename T> struct ClassTemplate {};

template <typename T>
void takesPointerToDependent(ClassTemplate<T> *) {}
```